src/externalized/DefaultContentManager.cc

```diff
@@ -191,34 +191,47 @@ DefaultContentManager::DefaultContentManager(GameVersion gameVersion,
 	m_movementCosts = NULL;
 }
 
-void DefaultContentManager::init()
+
+void DefaultContentManager::AddVFSLayer(VFS_ORDER order, const ST::string path, const bool throwOnError)
 {
-	SLOGI(ST::format("Vfs with stracciatella dir ({}) '{}'", VFS_ORDER_STRACCIATELLA, m_externalizedDataPath));
-	if (!Vfs_addDir(m_vfs.get(), VFS_ORDER_STRACCIATELLA, m_externalizedDataPath.c_str()))
+	bool succeeded = false;
+	if (Fs_isDir(path.c_str()))
 	{
-		RustPointer<char> err{getRustError()};
-		SLOGE(ST::format("DefaultContentManager::init '{}': {}", m_externalizedDataPath, err.get()));
-		throw std::runtime_error("Failed to add stracciatella dir");
+		SLOGI(ST::format("Adding directory to VFS ({}) '{}'", order, path));
+		succeeded = Vfs_addDir(m_vfs.get(), order, path.c_str());
+	}
+	else if (Fs_isFile(path.c_str()) && path.to_lower().ends_with(".slf"))
+	{
+		SLOGI(ST::format("Adding SLF archive to VFS ({}) '{}'", order, path));
+		succeeded = Vfs_addSlf(m_vfs.get(), order, path.c_str());
+	}
+	else
+	{
+		SLOGE("Unsupported file type.");
+		throw std::runtime_error("Failed to add to VFS");
 	}
 
-	SLOGI(ST::format("Vfs with vanilla dir ({}) '{}'", VFS_ORDER_VANILLA, m_dataDir));
-	if (!Vfs_addDir(m_vfs.get(), VFS_ORDER_VANILLA, m_dataDir.c_str()))
+	if (!succeeded)
 	{
-		RustPointer<char> err{getRustError()};
-		SLOGE(ST::format("DefaultContentManager::init '{}': {}", m_dataDir, err.get()));
-		throw std::runtime_error("Failed to add vanilla data dir");
+		RustPointer<char> err{ getRustError() };
+		SLOGE(ST::format("Failed to add to VFS '{}': {}", path, err.get()));
+		if (throwOnError)
+		{
+			throw std::runtime_error("Failed to add to VFS");
+		}
 	}
+}
 
+void DefaultContentManager::init()
+{
+	AddVFSLayer(VFS_ORDER::ASSETS_USERHOME, FileMan::joinPaths(m_userHomeDir, "data"));
```

Comment:
Should this be `"data"` or `"externalized"?`

Reply:
It is a new directory so it's up to us. The codebase treats externalized and data the same  I chose `data` because this directory can also contain binary assets.